gateway/api.go

```diff
@@ -579,6 +579,10 @@ func (gw *Gateway) handleAddOrUpdate(keyName string, r *http.Request, isHashed b
 
 	// add key hash for newly created key
 	if gw.GetConfig().HashKeys && r.Method == http.MethodPost {
+		if newSession.BasicAuthData.Password != "" {
+			response.Key = ""
+		}
+
 		if isHashed {
 			response.KeyHash = keyName
```

Comment:
According to the description, `response.Key=""` should be inside this `if` block. Can you please explain why you added above?

Reply:
it is already inside an if block 
```
gw.GetConfig().HashKeys && r.Method == http.MethodPost
```
gw.GetConfig().HashKeys && r.Method == http.MethodPost

Reply:
What if we make this one `if newSession.BasicAuthData.Password != "" {`, `if newSession.isBasicAuth() {` so that the code looks more readable?